File: javascript/002.js
function* fibonacci(f1, f2) {
  var n1 = f1 || 0,
      n2 = f2 || 1;

  while (true) {
    var current = n1;
    n1 = n2;
    n2 = current + n1;

    yield current;
  }
}

function sumEvenFibonacciNumbersBelow(n) {
  var sum      = 0,
      sequence = fibonacci(0, 1),
      current;

  do {
    current = sequence.next().value;
    if (current % 2 === 0) {
      sum += current;
    }
  } while (current < n);

  return sum;
}

var answer = sumEvenFibonacciNumbersBelow(4000000);
console.log(answer);

File: python/002.py
def fibonacci(f1 = 0, f2 = 1):
    while True:
        yield f1
        f1, f2 = f2, f1 + f2

def sumEvenFibonacciNumbersBelow(n):
    sum = 0

    for next in fibonacci(0, 1):
        if next < 4000000:
            if next % 2 == 0:
                sum += next
        else:
            return sum

if __name__ == "__main__":
    answer = sumEvenFibonacciNumbersBelow(4000000)
    print(answer)
